misc/systemd/ipfs.service

```diff
@@ -0,0 +1,31 @@
+[Unit]
+Description=ipfs p2p daemon
+After=network.target
+Requires=network.target
+
+[Service]
+Type=simple
+User=ipfs
+RestartSec=1
+Restart=always
+PermissionsStartOnly=true
+Nice=18
+StateDirectory=/var/lib/ipfs
+Environment=IPFS_PATH=/var/lib/ipfs
+Environment=HOME=/var/lib/ipfs
+LimitNOFILE=8192
+Environment=IPFS_FD_MAX=8192
+EnvironmentFile=-/etc/sysconfig/ipfs
+StandardOutput=journal
+WorkingDirectory=/var/lib/ipfs
+ExecStartPre=-adduser --system --group --home /var/lib/ipfs ipfs
+ExecStartPre=-mkdir /var/lib/ipfs
+ExecStartPre=-/bin/chown ipfs:ipfs /var/lib/ipfs
+ExecStartPre=-/bin/chmod u+rwx /var/lib/ipfs
+ExecStartPre=-chpst -u ipfs /usr/bin/env ipfs init --profile=badgerds
+ExecStartPre=-chpst -u ipfs /usr/bin/env ipfs config profile apply server
+ExecStartPre=-chpst -u ipfs /usr/bin/env ipfs config Datastore.StorageMax "4000GB"
+ExecStart=/usr/bin/env ipfs daemon --enable-namesys-pubsub --enable-pubsub-experiment --migrate=true
```

Comment:
Pass `--init` to initialize the repo if not already initialized. That way, we can skip the `ExecStartPre` command.